racket/collects/syntax/to-string.rkt

```diff
@@ -55,10 +55,14 @@
                  ((loop init-line!) i))]
               [(pair? (syntax-e c))
                (advance c init-line!)
-               (printf "(")
+               (define c-paren-shape (syntax-property c 'paren-shape))
+               (printf (format "~a" (or c-paren-shape #\()))
```

Comment:
`printf` already handles format strings, so this could be `(printf "~a" (or c-paren-shape #\( ))`.